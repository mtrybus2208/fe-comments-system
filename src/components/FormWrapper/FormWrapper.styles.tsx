import React from 'react';
import styled from 'styled-components';

export const FormWrapper: React.FunctionComponent = styled.div`
  background-color: #1b1e22;
  margin: 50px auto;
  max-width: 700px;
  width: 100%;
  flex: 1 1 100%;
`;

export const Body = styled.div`
  padding: 60px 25px;

  @media ${({ theme }) => theme.media.tablet} {
    padding: 60px 80px;
  }
`;

export const Top = styled.div`
  box-shadow: 0 17px 24px 0 rgba(0, 0, 0, 0.17),
    inset -1px 1.7px 0 0 rgba(255, 255, 255, 0.017);
  background-color: #20252b;
  padding: 15px;
  display: flex;
  justify-content: center;
  align-items: center;
  text-transform: uppercase;
  letter-spacing: 1.2;
`;
