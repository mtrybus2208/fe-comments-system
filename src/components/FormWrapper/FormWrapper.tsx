import React from 'react';
import * as S from './FormWrapper.styles';

export interface FormWrapperProps {
  name?: string;
  children: JSX.Element[] | JSX.Element;
}

const FormWrapper: React.SFC<FormWrapperProps> = ({ name, children }) => {
  return (
    <S.FormWrapper>
      <S.Top>{name}</S.Top>
      <S.Body>{children}</S.Body>
    </S.FormWrapper>
  );
};

export default FormWrapper;
