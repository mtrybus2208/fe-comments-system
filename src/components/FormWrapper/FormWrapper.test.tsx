import React from 'react';
import { shallow } from 'enzyme';

import FormWrapper from './FormWrapper';

describe('ResetPassword', () => {
  it('should render layout', () => {
    const wrapper = shallow(
      <FormWrapper name="foo">
        <span>bar</span>
      </FormWrapper>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
