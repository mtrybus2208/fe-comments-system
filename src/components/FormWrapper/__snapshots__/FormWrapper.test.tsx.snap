// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword should render layout 1`] = `ShallowWrapper {}`;
